lib/chat_api/widget_settings.ex

```diff
@@ -0,0 +1,118 @@
+defmodule ChatApi.WidgetSettings do
+  @moduledoc """
+  The WidgetSettings context.
+  """
+
+  import Ecto.Query, warn: false
+  alias ChatApi.Repo
+
+  alias ChatApi.WidgetSettings.WidgetSetting
+
+  @doc """
+  Returns the list of widget_settings.
+
+  ## Examples
+
+      iex> list_widget_settings()
+      [%WidgetSetting{}, ...]
+
+  """
+  def list_widget_settings do
+    Repo.all(WidgetSetting)
+  end
+
+  @doc """
+  Gets a single widget_setting.
+
+  Raises `Ecto.NoResultsError` if the Widget config does not exist.
+
+  ## Examples
+
+      iex> get_widget_setting!(123)
+      %WidgetSetting{}
+
+      iex> get_widget_setting!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_widget_setting!(id), do: Repo.get!(WidgetSetting, id)
+
+  def create_or_update(nil, params) do
+      create_widget_setting(params)
+  end
+
+  def create_or_update(id, params) do
```

Comment:
rather than looking up by id, we could just do a `find_by_account_id` (instead of `get_widget_setting!(id)`) -- otherwise it could be possible for someone to update the widget settings of a different account if they knew the id